src/wallet/wallet.cpp

```diff
@@ -2006,23 +1978,15 @@ CAmount CWalletTx::GetAvailableCredit(interfaces::Chain::Lock& locked_chain, boo
     if (pwallet == nullptr)
         return 0;
 
+    bool got_cache = filter == ISMINE_SPENDABLE || filter == ISMINE_WATCH_ONLY;
+    fUseCache &= got_cache;
```

Comment:
It seems pointless to overwrite the fUseCache argument. It would seem clearer just to write `fUseCache && allow_cache` in the one place the argument is used below to avoid having its value change in the middle of the function call.

Reply:
Good point, fixed.